[WinCairo] include secure cookies for secure WebSockets

mxschmitt opened the pull request:
#### 11421fd47f7db72214a7f810d5ecc3ad4dc8f1e4
<pre>
[WinCairo] include secure cookies for secure WebSockets
<a href="https://bugs.webkit.org/show_bug.cgi?id=251581">https://bugs.webkit.org/show_bug.cgi?id=251581</a>

Reviewed by Fujii Hironori.

Before this patch for wss:// connections, secure cookies were not
includes. This fixes it and includes them.

Canonical link: <a href="https://commits.webkit.org/259975@main">https://commits.webkit.org/259975@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/af31d23d018ce4ce868b6d76d55087228a62bbb4

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/106557 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/15573 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/39366 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/115742 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/115382 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/110466 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/17068 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/6801 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/98750 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/112325 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/12966 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95939 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/40525 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94844 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/27583 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/82246 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8801 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28935 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/9343 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/5967 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-pseudo/first-line-change-inline-color.html (failure)") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14961 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/48483 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10882 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/3704 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebCore/platform/network/curl/NetworkStorageSessionCurl.cpp:
```diff
@@ -53,12 +53,12 @@ static String defaultCookieJarPath()
 
     auto searchHTTPOnly = forHTTPHeader ? std::nullopt : std::optional<bool> { false };
-    auto secure = url.protocolIs("https"_s) ? std::nullopt : std::optional<bool> { false };
+    auto secure = (includeSecureCookies == IncludeSecureCookies::Yes || url.protocolIs("https"_s)) ? std::nullopt : std::optional<bool> { false };
```
I'm not an expert. But, looking at the code of WebCore and other ports, I think we should remove `|| url.protocolIs("https"_s)`.
https://github.com/WebKit/WebKit/blob/14d3cca9ff6908a009167387b63846c5db002d74/Source/WebCore/loader/CookieJar.cpp#L59

fujii commented on Source/WebCore/platform/network/curl/NetworkStorageSessionCurl.cpp:
```diff
@@ -135,10 +135,9 @@ HTTPCookieAcceptPolicy NetworkStorageSession::cookieAcceptPolicy() const
 {
-    // FIXME: This should filter secure cookies out if the caller requests it.
-    return { cookiesForSession(*this, firstParty, url, false), false };
+    return { cookiesForSession(*this, firstParty, url, false, includeSecureCookies), false };
```
The second argument is used for `document.setSecureCookiesAccessed`. It seems important. https://commits.webkit.org/192473@main It'd be nice, if you properly calculate the second argument in this time.

mxschmitt replied:
We can follow-up on this one since its not related to the bug report and I didn't find an easy way of doing it.

mxschmitt committed:
[WinCairo] include secure cookies for secure WebSockets
https://bugs.webkit.org/show_bug.cgi?id=251581

Reviewed by Fujii Hironori.

Before this patch for wss:// connections, secure cookies were not
includes. This fixes it and includes them.

Canonical link: https://commits.webkit.org/259975@main
